openhcl/virt_mshv_vtl/src/processor/mshv/arm64.rs

```diff
@@ -152,12 +155,25 @@ impl BackingPrivate for HypervisorBackedArm64 {
                 this.backing.next_deliverability_notifications;
         }
 
+        this.backing.last_vtl = None;
+
         let intercepted = this
             .runner
             .run()
             .map_err(|e| VpHaltReason::Hypervisor(UhRunVpError::Run(e)))?;
 
         if intercepted {
+            this.backing.last_vtl = Some(
+                hvdef::HvArm64InterceptMessageHeader::ref_from_prefix(
```

Comment:
No, this is wrong. Not all intercept messages start with this header.

(This is a bug in the legacy HCL.)

We need to get this from the register page.

Reply:
The register page doesn't exist on ARM. I confirmed with the hv team. The advice from them was to get it from the intercept message.

Reply:
Oh, I'm sorry, I looked too quickly and missed that this was ARM64.

Are we certain that all current and future messages on ARM will start with the header?

Reply:
I think we're certain that all current messages on ARM start with the header, so I can change it so that we only retrieve the last VTL for the current messages that we're handling.

We would still have the issue being discussed on the other thread though.